tools/header_schema.py

```diff
@@ -1,236 +1,98 @@
-    facility: Optional[str] = field(default=None)
-    proposalID: Optional[str] = field(default=None)
-    doi: Optional[str] = field(default=None)
-
-
-# class Polarization(str, enum.Enum):
-#     """The first symbol indicates the magnetisation direction of the incident beam.
-#     An optional second symbol indicates the direction of the scattered beam, if a spin analyser is present."""
-#
-#     unpolarized = "unpolarized"
-#     p = "+"
-#     m = "-"
-#     mm = "--"
-#     mp = "-+"
-#     pm = "+-"
-#     pp = "++"
-
-
-@dataclass
-class File:
-    file: str
-    timestamp: datetime.datetime
-
-
-@dataclass
-class Value:
-    magnitude: Union[float, List[float]]
-    unit: Optional[str] = field(default=None, metadata={"description": "SI unit string"})
-
-
-@dataclass
-class ValueRange:
-    min: float
-    max: float
-    unit: Optional[str] = field(default=None, metadata={"description": "SI unit string"})
-    steps: Optional[int] = None
-
-
-@dataclass
-class InstrumentSettings:
-    incident_angle: Union[Value, ValueRange]
-    wavelength: Union[Value, ValueRange]
-    polarization: Optional[Union[Literal['unpolarized', 'p', 'm', 'mm', 'mp', 'pm', 'pp']]] = field(
-        default='unpolarized',
-        metadata={
-            'description':
-                'Polarization described as unpolarized/ p / m / pp / pm / mp / mm / vector'
-        })
-
-
-@dataclass
-class Measurement:
-    instrument_settings: InstrumentSettings
-    data_files: List[Union[File, str]]
-    scheme: Optional[Union[
-        Literal[
-            "angle- and energy-dispersive",
-            "angle-dispersive",
-            "energy-dispersive",
-        ]
-    ]] = None
-
-
-@dataclass
-class Software:
-    """Description of the reduction software."""
-    name: str
-    version: Optional[str] = None
-    platform: Optional[str] = None
-
-
-@dataclass
-class Reduction:
-    software: Union[Software, str]
-    computer: Optional[str] = field(
-        default=None, metadata={'description': 'Computer used for reduction'}
-    )
-    call: Optional[str] = ""
-    timestamp: Optional[datetime.datetime] = field(
-        default=None,
-        metadata={
-            "description": "Timestamp string, formatted as ISO 8601 datetime"
-        })
-    creator: Optional[Person] = None
-    corrections: Optional[List[str]] = None
-    call: Optional[str] = field(
-        default=None, metadata={'description': 'The command line call used'})
-    script: Optional[str] = field(
-        default=None,
-        metadata={'description': 'Path to reduction script or notebook'})
-    binary: Optional[str] = field(
-        default=None,
-        metadata={'description': 'Path to full information file'})
-
-
-@dataclass
-class DataSource:
-    owner: Person
-    experiment: Experiment
-    sample: Sample
-    measurement: Measurement
-
-
-@dataclass
-class Column:
+import os
+from typing import Dict, List
+from copy import deepcopy
+
+
+os.environ["GENERATE_SCHEMA"] = "True"
```

Comment:
I think instead of using this variable you could just overwrite the `dataclasses.dataclass` method here so it gets imported in orsopy.fileio.base in the modified version.

Reply:
You could do something like:
```python
import dataclasses
_dataclass = dataclasses.dataclass

...

dataclasses.dataclass = functools.partial(_dataclass, config=PydanticConfig)
```

Reply:
The approach you've described doesn't work - but I can try to find something that does.  I agree that having the logic to generate the schemas would ideally be separated from the orsopy base classes.
  
I do feel that schema generation is not similar to unit tests though; the schemas are a language-independent representation of the standard that can be (are) used in other contexts, while the unit tests have no meaning outside of orsopy.